R/backend-mssql.R

```diff
@@ -372,6 +372,18 @@ simulate_mssql <- function(version = "15.0") {
       get_day = function(x) {
         sql_expr(DATEPART(DAY, !!x))
       },
+      date_count_between = function(start, end, precision, ..., n = 1L){
+
+        check_dots_empty()
+        if (precision != "day") {
+          cli::cli_abort('The only supported value for {.arg precision} on SQL backends is "day"')
```

Comment:
```suggestion
          cli_abort("{.arg precision} must be {.val day} on SQL backends.")
```

A couple notes here related to the [tidyverse style guide](https://style.tidyverse.org/errors.html) for error messages:

* `cli_abort()` is imported into the package namespace, so we don't need to namespace it!
* Using `.val` markup will allow us to use double-quotes.
* Added a period.
* Began the message with the problematic argument and used "must".

Could you apply these changes elsewhere in the PR? After those are made, I think this PR should be in a great spot.